source/tagged-union.d.ts

```diff
@@ -0,0 +1,41 @@
+/**
+Utility type to create a tagged union with given tag
+
+Use-case: A shorter way to declare tagged unions with multiple members.
+
+@example
+```
+import type {TaggedUnion} from 'type-fest';
+
+type EventMessage = TaggedUnion<
+  'tag',
+  {
+    NavigationStateChanged: {
```

Comment:
Use tab-indentation.

Reply:
I reformatted the comment with xo. Should have tabs now :)